### PR TITLE
Normalize view direction in universal

### PR DESCRIPTION
**Pull request by elizabeth-legros**
Needed to have the view direction node return a normalized value in Shader Graph. Created templates in the Universal package to achieve this. 

Ran all Shader Graph tests locally and verified normalized view direction in editor

### CODE REVIEW

**Review by cdxntchou — APPROVED** [review]

**Review by phi-lira — CHANGES_REQUESTED** [review]
Missing changelog. Otherwise LGTM.

**Review by LandonTownsendUnity — CHANGES_REQUESTED** [review]
I would like documentation updates both in the View Direction node documentation and the upgrade guide.

File: com.unity.shadergraph/Documentation~/View-Direction-Node.md
Comment (cdxntchou) [inline com.unity.shadergraph/Documentation~/View-Direction-Node.md:7]: Do you know if the camera position output in (local) World space in HDRP?  (i.e. 0,0,0) ?

File: com.unity.shadergraph/Editor/Data/Nodes/CodeFunctionNode.cs
Comment (cdxntchou) [inline com.unity.shadergraph/Editor/Data/Nodes/CodeFunctionNode.cs:283]: Is this change related?  Or just setting up nice defaults?

**Review by cdxntchou — APPROVED** [review]

File: com.unity.shadergraph/Editor/Data/Nodes/CodeFunctionNode.cs
Comment (elizabeth-legros) [reply com.unity.shadergraph/Editor/Data/Nodes/CodeFunctionNode.cs:283]: change related because it wasnt getting passed in before (which would default to false when making the slots) so I just wanted to keep it as consistent as possible

File: com.unity.shadergraph/Documentation~/View-Direction-Node.md
Comment (elizabeth-legros) [reply com.unity.shadergraph/Documentation~/View-Direction-Node.md:7]: Yeah, that was something i wan going to have to double check in HDRP. I think you are right tho, and will address it

Comment (elizabeth-legros) [reply com.unity.shadergraph/Documentation~/View-Direction-Node.md:7]: This wording is getting changed tho, since weve added the new node

**Review by phi-lira — APPROVED** [review]

**Comment by phi-lira** [comment]
@LandonTownsendUnity the feedback was addressed can you look at this PR again?

**Comment by elizabeth-legros** [comment]
> @LandonTownsendUnity the feedback was addressed can you look at this PR again?

We are also waiting on updated Documentation for the node before we can merge

**Review by LandonTownsendUnity — CHANGES_REQUESTED** [review]
I'm OK with the "view vector" object mode behavior being different because the old view direction behavior was normalized and that doesn't make sense for a "view vector" node. However it should be stated in the upgrade guide/documentation that you shouldn't swap out the node if it's in object space.

The behavior of "View Vector" is different from the old behavior of View Direction when in orthographic camera mode, in all spaces. This needs to be fixed. Even aside from upgrade problems, "view vector" being the vector from the position to the camera doesn't make sense in orthographic mode.

**Review by LandonTownsendUnity — CHANGES_REQUESTED** [review]
View Vector needs to be fixed in HDRP - currently uses camera relative World transformations instead of Absolute World

Edit: actually hold on, let me check that

**Review by LandonTownsendUnity — COMMENTED** [review]
Changelog also needs to be updated with the fact that the View Vector node was added.

**Review by LandonTownsendUnity — APPROVED** [review]
Now that the changelog is updated, I will approve.

**Review by cdxntchou — APPROVED** [review]

**Pull request merged**
